jade_templates/includes/dashboard.js

```diff
@@ -105,35 +132,81 @@ ChangesGraph.prototype.updateSocketGraph = function(result){
     this._chart.draw(1000);
 };
 
-var mostActiveArticles = new ChangesGraph('#active_articles', 'Article', 'title', (d) => d.message.title);
-mostActiveArticles.loadSocketGraph();
-
-var mostActiveUsers = new ChangesGraph('#active_users', 'User','user', (d) => d.message.user);
-mostActiveUsers.loadSocketGraph();
-
-var mostUsedActionTypes = new ChangesGraph('#active_action_types', 'Action Type','type', (d) => d.message.type);
-mostUsedActionTypes.loadSocketGraph();
-
-var mostUsedComments = new ChangesGraph('#active_comments', 'Comment','comment', (d) => d.message.comment);
-mostUsedComments.loadSocketGraph();
-
-var publisher = new DataPublisher({
-    url: '/?dash_socketdata=1',
-    timeout: 5000
-},[
-    function(data){
-        mostActiveArticles.updateSocketGraph(data);
-    },
-    function(data){
-        mostActiveUsers.updateSocketGraph(data);
-    },
-    function(data){
-        mostUsedActionTypes.updateSocketGraph(data);
-    },
-    function(data){
-        // XXX Hack to improve display of empty comments
-        data.data.forEach((d) => d.message.comment = d.message.comment || '(Empty)');
-        mostUsedComments.updateSocketGraph(data);
+// d3.json('/?dash_config=1', function(err, data){
+(function(err, data){
+    var err = null;
+    if(err) {
+        alert('Error - could not show dashboard. See log for more details.');
+        throw err;
+    }
+
+    var config = data.config;
+    
+    if(config && config.graphList && config.graphList.length) {
+        if( ! window.graphList) window.graphList = {};
+
+        var mainGraph = d3.select('#main_graph_area');
+        var graphSubList = [];
+        for(var gi in config.graphList) {
+            var graph = config.graphList[gi],
+                type = graph.type.split('.'),
+                gClass = window;
+
+            for(var i in type) {
+                gClass = gClass[type[i]];
+            }
+
+            var chartElement = mainGraph.append('div').classed('chart', true);
+            var argsCopy = graph.arguments.slice();
+            argsCopy.unshift(null)
+            var newGraph = new (Function.prototype.bind.apply(gClass, argsCopy));
+            newGraph.loadGraph(chartElement);
+            graphSubList.push(newGraph.updateGraph.bind(newGraph));
+            window.graphList['graph_' + gi] = newGraph;
+        }
+        var publisher = new DataPublisher({
+            url: '/?dash_socketdata=1',
+            timeout: 5000,
+            timeRange: (60*60*24)
```

Comment:
Note that this extends the graph time range to 24 hours. Let me know if that's too much.